xwiki-platform-core/xwiki-platform-icon/xwiki-platform-icon-test/xwiki-platform-icon-test-docker/src/test/it/org/xwiki/icon/test/ui/IconThemesAdminIT.java

```diff
@@ -0,0 +1,118 @@
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.icon.test.ui;
+
+import java.util.Arrays;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.openqa.selenium.By;
+import org.xwiki.administration.test.po.AdministrablePage;
+import org.xwiki.administration.test.po.AdministrationPage;
+import org.xwiki.administration.test.po.ThemesAdministrationSectionPage;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.test.docker.junit5.UITest;
+import org.xwiki.test.ui.TestUtils;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Functional tests of the icon theme Admin features.
+ *
+ * @version $Id$
+ * @since 16.10.0RC1
+ */
+@UITest
+class IconThemesAdminIT
+{
+    private static final String SILK_THEME = "Silk";
+
+    private static final String FA_THEME = "Font Awesome";
+
+    @Test
+    void validateIconThemeFeatures(TestUtils setup, TestInfo info)
+    {
+        setup.loginAsSuperAdmin();
+
+        // Validate setting a icon theme from the wiki Admin UI
+        validateSetThemeFromWikiAdminUI(setup);
+
+        // Validate setting a icon theme from the page Admin UI for the page and its children
+        validateSetThemeFromPageAdminUI(setup, info);
+    }
+
+    private void validateSetThemeFromWikiAdminUI(TestUtils setup)
+    {
+        // Go to the Theme Admin UI to verify we can set a new theme from there using the select control
+        AdministrationPage administrationPage = AdministrationPage.gotoPage();
+        ThemesAdministrationSectionPage presentationAdministrationSectionPage = administrationPage.clickThemesSection();
+        // We expect the default icon to be Silk
+        assertFalse(setup.isUsingFA());
+
+        // Set the new icon theme as the active theme
+        presentationAdministrationSectionPage.setIconTheme(FA_THEME);
+        assertEquals(FA_THEME, presentationAdministrationSectionPage.getCurrentIconTheme());
+        presentationAdministrationSectionPage.clickSave();
+
+        // Verify that the icon theme has been applied.
+        assertTrue(setup.isUsingFA());
+
+        // Switch back to Silk
+        administrationPage = AdministrationPage.gotoPage();
+        presentationAdministrationSectionPage = administrationPage.clickThemesSection();
```

Comment:
Why navigate using the UI again? Anytime you do this, it costs time to the test vs direct nav.

Now here I don't understand why you're navigating at all.